rest-api-spec/src/main/resources/rest-api-spec/api/ml.start_deployment.json

```diff
@@ -0,0 +1,30 @@
+{
+  "ml.start_deployment":{
+    "documentation":{
+      "url":"https://www.elastic.co/guide/en/elasticsearch/reference/current/ml-start-deployment.html",
+      "description":"Start a trained model deployment."
+    },
+    "stability":"experimental",
+    "visibility":"public",
+    "headers":{
+      "accept": [ "application/json"],
+      "content_type": ["application/json"]
+    },
+    "url":{
+      "paths":[
+        {
+          "path":"/_ml/trained_models/deployment/{model_id}/_start",
```

Comment:
Other paths for trained model APIs start with `/_ml/trained_models/{model_id}`, would it be more consistent to use `/_ml/trained_models/{model_id}/deployment/[_start|_stop]`?

Reply:
I switched the path to `/_ml/trained_models/{model_id}/deployment/_XXX` and added the _infer endpoint spec (ml.infer_trained_model_deployment.json)